packages/react/src/themes/teams/components/Status/statusStyles.ts

```diff
@@ -1,81 +1,64 @@
-import { pxToRem, SizeValue } from '../../../../lib'
-import { ComponentSlotStylesInput, ICSSInJSStyle } from '../../../types'
 import { StatusProps } from '../../../../components/Status/Status'
 import { StatusVariables } from './statusVariables'
+import { ComponentSelectorsAndStyles } from 'src/themes/types'
+import { backportComponentStyle } from 'src/lib/resolveComponentRules'
 
-const getBackgroundColor = (state: string, variables: StatusVariables) => {
-  switch (state) {
-    case 'success':
-      return variables.successBackgroundColor
-    case 'info':
-      return variables.infoBackgroundColor
-    case 'warning':
-      return variables.warningBackgroundColor
-    case 'error':
-      return variables.errorBackgroundColor
-    case 'unknown':
-    default:
-      return variables.defaultBackgroundColor
-  }
-}
+const statusStyles: ComponentSelectorsAndStyles<StatusProps, StatusVariables> = v => ({
+  root: [
+    [
+      null,
+      {
+        display: 'inline-flex',
+        alignItems: 'center',
+        justifyContent: 'center',
+        verticalAlign: 'middle',
+        width: v.medium,
+        height: v.medium,
+        color: v.defaultTextColor,
+        backgroundColor: v.defaultBackgroundColor,
+        borderRadius: '9999px',
+      },
+    ],
 
-const getTextColor = (state: string, variables: StatusVariables) => {
-  switch (state) {
-    case 'success':
-      return variables.successTextColor
-    case 'info':
-      return variables.infoTextColor
-    case 'warning':
-      return variables.warningTextColor
-    case 'error':
-      return variables.errorTextColor
-    case 'unknown':
-    default:
-      return variables.defaultTextColor
-  }
-}
+    //
+    // Colors
+    //
+    [{ color: 'red' }, { backgroundColor: v.backgroundRed }],
+    [{ color: 'orange' }, { backgroundColor: v.backgroundOrange }],
```

Comment:
Component parts are now arrays of tuples.  The tuples are `[selector, style]`.  If the props object matches the component's props, then the style is applied.  Matched styles are deep merged from top to bottom.

This makes this syntax work much like a CSS sheet:

```jsx
    [{ color: 'red' }, { backgroundColor: v.backgroundRed }],
    [{ color: 'orange' }, { backgroundColor: v.backgroundOrange }],
```
Which is equivalent to this SCSS:
```css
    .ui-status.red { background-color: $backgroundRed; }
    .ui-status.orange { background-color: $backgroundOrange; }
```

The selector information will allow us to transform props into CSS selectors as well.